### PR TITLE
Initial implementation of httpclient 5

### PR DESCRIPTION
**Pull request by ryanjbaxter**

### CODE REVIEW

File: spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/HttpClientConfiguration.java [hunk 1]
```diff
@@ -33,8 +33,34 @@
 public class HttpClientConfiguration {
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnProperty(name = "spring.cloud.httpclientfactories.apache.enabled", matchIfMissing = true)
+	@ConditionalOnProperty(name = "spring.cloud.httpclientfactories.apache5.enabled", matchIfMissing = true)
```
Comment (spencergibb) [inline spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/HttpClientConfiguration.java:36]: I probably leave `apache`

Comment (spencergibb) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/HttpClientConfiguration.java:36]: Oh, you've got both?

Comment (ryanjbaxter) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/HttpClientConfiguration.java:36]: Right implementations for both, depending on whats on the classpath

Comment (ryanjbaxter) [reply spring-cloud-commons/src/main/java/org/springframework/cloud/commons/httpclient/HttpClientConfiguration.java:36]: They would have to put apache4 on the classpath AND enable to property enabled

**Pull request closed without merging**